File: app/src/main/java/com/example/gymsessionsstopwatch_wearos/local_files/LocalFileManager.java
package com.example.gymsessionsstopwatch_wearos.local_files;

import android.content.Context;
import com.google.gson.Gson;

import com.example.gymsessionsstopwatch_wearos.sessions.WorkoutSession;
import com.google.gson.reflect.TypeToken;

import java.io.File;
import java.io.FileInputStream;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.ArrayList;
import java.util.List;

public class LocalFileManager {
    private static final String FILE_NAME = "workout_sessions.json";

    public static void saveSessionsToFile(Context context, List<WorkoutSession> sessions) {
        Gson gson = new Gson();
        String jsonString = gson.toJson(sessions);

        try (FileOutputStream fos = context.openFileOutput(FILE_NAME, Context.MODE_PRIVATE);
             OutputStreamWriter writer = new OutputStreamWriter(fos)) {
            writer.write(jsonString);
        } catch (Exception e) {
            e.printStackTrace();
        }

    }

    public static List<WorkoutSession> loadSessionFromFile(Context context) {
        FileInputStream fis = null;
        try {
            File file = new File(context.getFilesDir(), FILE_NAME);
            if (!file.exists()) {
                return new ArrayList<>();
            }

            fis = context.openFileInput(FILE_NAME);
            InputStreamReader reader = new InputStreamReader(fis);
            Gson gson = new Gson();

            return gson.fromJson(reader, new TypeToken<List<WorkoutSession>>() {}.getType());

        } catch (Exception e) {
            e.printStackTrace();
            return null;
        } finally {
            if (fis != null) {
                try {
                    fis.close();
                } catch (IOException e) {
                    e.printStackTrace();
                }
            }
        }
    }
}

File: app/src/main/java/com/example/gymsessionsstopwatch_wearos/local_files/LocalFileManager.kt
package com.example.gymsessionsstopwatch_wearos.local_files

import android.content.Context
import com.example.gymsessionsstopwatch_wearos.sessions.WorkoutSession
import com.google.gson.Gson
import com.google.gson.reflect.TypeToken
import java.io.*

object LocalFileManager {
    private const val FILE_NAME = "workout_sessions.json"

    fun saveSessionsToFile(context: Context, sessions: List<WorkoutSession>) {
        val gson = Gson()
        val jsonString = gson.toJson(sessions)
        try {
            context.openFileOutput(FILE_NAME, Context.MODE_PRIVATE).use { fos ->
                OutputStreamWriter(fos).use { writer ->
                    writer.write(jsonString)
                }
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    fun loadSessionsFromFile(context: Context): List<WorkoutSession>? {
        try {
            val file = File(context.filesDir, FILE_NAME)
            if (!file.exists()) {
                return arrayListOf()
            }
            context.openFileInput(FILE_NAME).use { fis ->
                InputStreamReader(fis).use { reader ->
                    val gson = Gson()
                    return gson.fromJson(reader, object : TypeToken<List<WorkoutSession>>() {}.type)
                }
            }
        } catch (e: Exception) {
            e.printStackTrace()
            return null
        }
    }
}
